docs/rules/no-param-reassign.md

```diff
@@ -74,6 +74,25 @@ function foo(bar) {
 }
 ```
 
+Examples of **correct** code for the `{ "props": true }` option with `"ignorePropertyAssignmentsFor"` set:
+
+```js
+/*eslint no-param-reassign: ["error", { "props": true, "ignorePropertyAssignmentsFor": ["bar"] }]*/
+
+function foo(bar) {
+    bar.prop = "value";
+}
+
+function foo(bar) {
```

Comment:
If we want to allow things other than assignment, I think the option should be named `ignorePropertyModificationFor` instead